lib/arInstall.class.php

```diff
@@ -0,0 +1,421 @@
+        }
+    }
+
+    public static function createDatabasesYml()
+    {
+        $databasesYmlPath = sfConfig::get('sf_config_dir').'/databases.yml';
+
+        // Read databases.yml contents from existing databases.yml,
+        // databases.yml.tmpl (for a Subversion checkout), or symfony
+        // skeleton databases.yml, whichever is found first.
+        $databasesYmlPaths = [];
+        $databasesYmlPaths[] = $databasesYmlPath;
+        $databasesYmlPaths[] = $databasesYmlPath.'.tmpl';
+        $databasesYmlPaths[] = sfConfig::get('sf_lib_dir')
+            .'/task/generator/skeleton/project/config/databases.yml';
+
+        foreach ($databasesYmlPaths as $path) {
+            if (false !== $content = file_get_contents($path)) {
+                break;
+            }
+        }
+
+        if (false === file_put_contents($databasesYmlPath, $content)) {
+            throw new Exception(
+                "Can't write configuration file {$databasesYmlPath}"
+            );
+        }
+    }
+
+    public static function createPropelIni()
+    {
+        $propelIniPath = sfConfig::get('sf_config_dir').'/propel.ini';
+
+        // Read propel.ini contents from existing propel.ini,
+        // propel.ini.tmpl (for a Subversion checkout), or symfony
+        // skeleton propel.ini, whichever is found first.
+        $propelIniPaths = [];
+        $propelIniPaths[] = $propelIniPath;
+        $propelIniPaths[] = $propelIniPath.'.tmpl';
+        $propelIniPaths[] = sfConfig::get('sf_lib_dir')
+            .'/task/generator/skeleton/project/config/propel.ini';
+
+        foreach ($propelIniPaths as $path) {
+            if (false !== $content = file_get_contents($path)) {
+                break;
+            }
+        }
+
+        if (false === file_put_contents($propelIniPath, $content)) {
+            throw new Exception(
+                "Can't write configuration file {$propelIniPath}"
+            );
+        }
+    }
+
+    public static function createSettingsYml()
+    {
+        $settingsYmlPath = sfConfig::get('sf_app_config_dir').'/settings.yml';
+
+        // Read settings.yml contents from existing settings.yml,
+        // settings.yml.tmpl (for a Subversion checkout), or symfony
+        // skeleton settings.yml, whichever is found first.
+        $settingsYmlPaths = [];
+        $settingsYmlPaths[] = $settingsYmlPath;
+        $settingsYmlPaths[] = $settingsYmlPath.'.tmpl';
+        $settingsYmlPaths[] = sfConfig::get('sf_lib_dir')
+            .'/task/generator/skeleton/app/app/config/settings.yml';
+
+        foreach ($settingsYmlPaths as $path) {
+            if (false !== $content = file_get_contents($path)) {
+                break;
+            }
+        }
+
+        // Generate and set CSRF secret
+        $content = str_replace(
+            'change_me',
+            bin2hex(openssl_random_pseudo_bytes(16)),
+            $content
+        );
+        if (false === file_put_contents($settingsYmlPath, $content)) {
```

Comment:
I don't think that's the case. Or do you mean before control statements? In which case, `if` it's not included in PphCsFixer ruleset:

https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/doc/rules/whitespace/blank_line_before_statement.rst#rule-sets

Reply:
Thanks for the link - Which ruleset are we using?

Reply:
We use the `@PhpCsFixer` ruleset with a single exception to avoid inconsistencies in templates:

https://github.com/artefactual/atom/blob/qa/2.x/.php_cs#L24-L25